tests/evm/precompiles/test_ecRecover.py

```diff
@@ -0,0 +1,173 @@
+import pytest
+
+from typing import List
+from common import CallContext, rand_fq
+from eth_keys import keys  # type: ignore
+from eth_utils import keccak
+from zkevm_specs.evm_circuit import (
+    Bytecode,
+    CallContextFieldTag,
+    ExecutionState,
+    Precompile,
+    RWDictionary,
+    StepState,
+    Tables,
+    verify_steps,
+)
+from zkevm_specs.evm_circuit.execution.precompiles.ecrecover import SECP256K1N
+from zkevm_specs.util import (
+    Word,
+    FQ,
+)
+from zkevm_specs.evm_circuit.table import SigTableRow
+
+
+def gen_testing_data():
+    # basic
+    msg = "Hello World!"
+    sk = keys.PrivateKey(rand_fq().n.to_bytes(32, "little"))
+    address = sk.public_key.to_canonical_address()
+    msg_hash = keccak(bytes(msg, "utf-8"))
+    sig = sk.sign_msg_hash(msg_hash)
+    v = sig.v
+    r = sig.r
+    s = sig.s
+
+    # successful case
+    normal = [CallContext(), msg_hash, v, r, s, address]
+
+    # failure cases
+    zero_addr = [CallContext(), msg_hash, v, r, s, bytes(0)]
+    sig_r_over_ub = [CallContext(), msg_hash, v, SECP256K1N, s, bytes(0)]
+    sig_s_over_ub = [CallContext(), msg_hash, v, r, SECP256K1N, bytes(0)]
+    sig_r_one = [CallContext(), msg_hash, v, 1, s, bytes(0)]
+    sig_s_one = [CallContext(), msg_hash, v, r, 1, bytes(0)]
```

Comment:
Are we testing the `non-zero` case or just random `r` `s` (if latter I'd suggest to use random value)

Reply:
I was testing the lower bound of r and s and I thought the criteria is larger than `1`. Corrected both of them to `0` and fixed in [b517a95](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/495/commits/b517a957e50b0da2303856db80033f33e1cda63c)